### PR TITLE
Return specific error code when node is not found

### PR DESCRIPTION
**Pull request by ivanpauno**
Addresses https://github.com/ros2/ros2cli/pull/322#issuecomment-525364023

### CODE REVIEW

File: rmw_fastrtps_shared_cpp/src/rmw_node_info_and_types.cpp [hunk 1]
```diff
@@ -85,7 +85,7 @@ rmw_ret_t __get_guid_by_name(
         kLoggerTag,
         "Unable to find GUID for node: %s", node_name);
       RMW_SET_ERROR_MSG("Unable to find GUID for node ");
-      return RMW_RET_ERROR;
+      return RMW_RET_NON_EXISTENT_NODE_NAME;
```
Comment (dirk-thomas) [inline rmw_fastrtps_shared_cpp/src/rmw_node_info_and_types.cpp]: Is the caller still expected to clear the error message? Or with the explicit return code the error message could be removed?

Comment (ivanpauno) [reply rmw_fastrtps_shared_cpp/src/rmw_node_info_and_types.cpp]: We usually set the error message when returning an explicit return code, so I will follow that.
Deleting the logging of the error, as is unnecessary.

File: rmw_fastrtps_shared_cpp/src/rmw_node_info_and_types.cpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ rmw_ret_t __get_guid_by_name(
   const char * node_namespace, GUID_t & guid)
 {
   auto impl = static_cast<CustomParticipantInfo *>(node->data);
-  if (strcmp(node->name, node_name) == 0) {
+  if (strcmp(node->name, node_name) == 0 && strcmp(node->namespace_, node_namespace) == 0) {
```
Comment (ivanpauno) [inline rmw_fastrtps_shared_cpp/src/rmw_node_info_and_types.cpp]: Realized of this preexisting bug when uncommenting some tests in `rcl` `test_graph.cpp`.

**Comment by dirk-thomas** [comment]
> You might consider curating the changes into two commits

The unrelated change should even go into a separate PR imo. The change to introduce a new return code is already huge (across many repos). It shouldn't be combined with other changes.

**Comment by ivanpauno** [comment]
> > You might consider curating the changes into two commits
> 
> The unrelated change should even go into a separate PR imo. The change to introduce a new return code is already huge (across many repos). It shouldn't be combined with other changes.

I needed to add name and namespace validation in `rcl`, because if not some function calls in tests started returning `RCL_RET_NODE_NAME_NON_EXISTENT` when they shouldn't. After that, the PR in `rcl` went big.
Some parts of the tests where also commented, supposedly because we didn't have name/namespace validation, so I uncommented them. After that I realized that the error was this, and I corrected it.

I can comment again the problematic parts of the code in `rcl`, undo this extra change, and send them in a new PR. But I cannot delete the adding of node/namespace validation in `rcl` PR, because I wouldn't get green CI (and that's much bigger than this correction).

I think that splitting this PR in two commits (doing rebase and merge) is the easier way to go.

**Comment by ivanpauno** [comment]
I've already sent and merged the unrelated fix in a separated PR https://github.com/ros2/rmw_fastrtps/pull/316

**Pull request merged**